static/app/stores/pageFiltersStore.tsx

```diff
@@ -21,6 +21,12 @@ interface CommonState {
    * The current page filter selection
    */
   selection: PageFilters;
+  /**
+   * Whether to save changes to local storage. This setting should be page-specific:
+   * most pages should have it on (default) and some, like Dashboard Details, need it
+   * off.
+   */
+  shouldPersist: boolean;
```

Comment:
Decided to call this `shouldPersist` rather than `canPin`/`shouldPin`… because I'll soon be removing the pinning mechanism from the store. By default all filters will persist (`shouldPersist` will be `true`), and persistence can be disabled by setting `shouldPersist` to `false`.